### PR TITLE
12.0 mig project_task_pull_request

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa

### CODE REVIEW

File: project_task_pull_request/views/project_task_pull_request_view.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+    <record id="view_task_form2" model="ir.ui.view">
+        <field name="name">project.task.form.pr</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_form2"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='sequence']" position="before">
```
Comment (nikul-serpentcs) [inline project_task_pull_request/views/project_task_pull_request_view.xml]: @ernestotejeda IMO here direct use `<field name="sequence" position="before">`

File: project_task_pull_request/tests/test_project_task_pull_request.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+from odoo.exceptions import ValidationError
+
+
+class TestProjectTaskPullRequest(TransactionCase):
+    post_install = True
+    at_install = False
+
+    def setUp(self):
+        super(TestProjectTaskPullRequest, self).setUp()
+        self.project_1 = self.env['project.project'].create({
```
Comment (nikul-serpentcs) [inline project_task_pull_request/tests/test_project_task_pull_request.py]: `self.env['project.project']` use multipal time, so use like  global variable

File: project_task_pull_request/tests/test_project_task_pull_request.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            'name': 'Test Project',
+            'pr_required_states': [(4, 6, 0)],
+            'key': 'ABC',
+        })
+        self.project_2 = self.env['project.project'].create({
+            'name': 'Test Project',
+            'pr_required_states': [(4, 6, 0), (4, 5, 0)],
+            'key': 'DEF',
+        })
+        self.task_1 = self.env['project.task'].create({
```
Comment (nikul-serpentcs) [inline project_task_pull_request/tests/test_project_task_pull_request.py]: same as here

File: project_task_pull_request/tests/__init__.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# Copyright 2017 Specialty Medical Drugstore
```
Comment (nikul-serpentcs) [inline project_task_pull_request/tests/__init__.py]: remove copyright from `__init__` file

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

**Review by chienandalu — APPROVED** [review]
:+1: Tested on runbot. Thanks!

**Review by rodrigets — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@alexey-pelykh it seems `project_key` module is breaking these tests. Is it possible to disable default behavior when testing other modules?

**Comment by alexey-pelykh** [comment]
@pedrobaeza the check is based on SQL constraint https://github.com/OCA/project/blob/cd712c91d04dc1d52f11a4b34a2a8e9c542f8d92/project_key/models/project_project.py#L26
the only way to workaround it is to ensure https://github.com/OCA/project/blob/cd712c91d04dc1d52f11a4b34a2a8e9c542f8d92/project_key/models/project_project.py#L136
is generating unique key by making search
(in other words, improvement for `project_key`)

or alter the tests to be "Test Project A", "Test Project B", etc.

**Comment by pedrobaeza** [comment]
Uhm, I don't know exactly how it works the module, but indeed it seems a problem for each module in the repository. I'll check further later.

**Comment by pedrobaeza** [comment]
Fixed changing tests with different tasks/projects name.

**Pull request merged**